feat: due date reminders

Agrendalath opened the pull request:
Backport of https://github.com/openedx/edx-platform/pull/32826.

Private-ref: [BB-7407](https://tasks.opencraft.com/browse/BB-7407)

Agrendalath committed:
feat: allow disabling user tours

A similar waffle flag was dropped in 20386337.

navinkarkera committed:
feat: due date reminders

Adds a command to send reminder emails for self-paced courses with relative
due dates.

navinkarkera committed:
fix: update migrations from 5816a03 for Palm

kaustavb12 commented:
Sandbox deploy request received. Deployment will start soon.

kaustavb12 commented:
Sandbox deployment started.

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox destroy

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

gabor-boros commented:
/grove sandbox update

kaustavb12 commented:
/grove sandbox update

open-craft-grove commented:
Sandbox update request received. Deployment will start soon.

open-craft-grove commented:
Sandbox deployment started.

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at [pr-583-0e7f8a.staging.do.opencraft.hosting](https://pr-583-0e7f8a.staging.do.opencraft.hosting)
Sandbox Studio is available at [studio.pr-583-0e7f8a.staging.do.opencraft.hosting](https://studio.pr-583-0e7f8a.staging.do.opencraft.hosting)

open-craft-grove commented:
Sandbox deployment started.

open-craft-grove commented:
Sandbox deployment successful.

Sandbox LMS is available at [pr-583-0e7f8a.staging.do.opencraft.hosting](https://pr-583-0e7f8a.staging.do.opencraft.hosting)
Sandbox Studio is available at [studio.pr-583-0e7f8a.staging.do.opencraft.hosting](https://studio.pr-583-0e7f8a.staging.do.opencraft.hosting)

open-craft-grove commented:
Sandbox destroy request received.

open-craft-grove commented:
Sandbox destroy request received.

Agrendalath committed:
feat: handle countries not recognized by geoip2 DB

The `geoip2` library returns `None` when it only identifies the continent
related to the IP address.
